src/main/kotlin/com/tinder/StateMachine.kt

```diff
@@ -51,7 +51,14 @@ class StateMachine<STATE : Any, EVENT : Any, SIDE_EFFECT : Any> private construc
     private fun STATE.getDefinition() = graph.stateDefinitions
         .filter { it.key.matches(this) }
         .map { it.value }
-        .firstOrNull() ?: error("Missing definition for state ${this.javaClass.simpleName}!")
+        .also { if (it.isEmpty()) error("Missing definition for state ${this.javaClass.simpleName}!") }
+        .fold(Graph.State<STATE, EVENT, SIDE_EFFECT>()) { acc, state ->
```

Comment:
I like this change, this also works when we want to have a generic transition

```
private val stateMachine = StateMachine.create<String, Int, Nothing> {
                initialState(STATE_A)
                state(STATE_A) {
                    on(EVENT_1) {
                        transitionTo(STATE_B)
                    }
                }
                state(STATE_B) {
                    on(EVENT_2) {
                        transitionTo(STATE_A)
                    }
                }
                state<String> {
                    on(EVENT_3) {
                        transitionTo(STATE_C)
                    }
                }
            }
```
for example, this one, having `state<String>` or `state(any())` we can move from any state to another state with the same event.